### PR TITLE
Rotate credentials by accounts that are in Ready state

### PR DESCRIPTION
**Pull request by rafael-azevedo**
This PR updates the RotateConsoleCredentials and RotateCredentials functions so that they pull that lastest account CR data before updating.

### CODE REVIEW

**Comment by rafael-azevedo** [comment]
This PR has been tested

File: pkg/controller/account/credentials_rotator.go [hunk 1]
```diff
@@ -78,6 +79,17 @@ func (r *ReconcileAccount) RotateCredentials(reqLogger logr.Logger, awsSetupClie
...
 
+	// Get latest version of the account
+	accountObjectKey, err := client.ObjectKeyFromObject(account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get name and namespace of Acccount object")
+	}
+
+	err = r.Client.Get(context.TODO(), accountObjectKey, account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get updated Acccount object after status update")
```
Comment (jharrington22) [inline pkg/controller/account/credentials_rotator.go]: We are not updating the status anywhere above here?

File: pkg/controller/account/credentials_rotator.go [hunk 2]
```diff
@@ -178,7 +190,18 @@ func (r *ReconcileAccount) RotateConsoleCredentials(reqLogger logr.Logger, awsSe
...
-	// Set `status.RotateCredentials` to false now that they ahve been updated
+	// Get latest version of the account
+	accountObjectKey, err := client.ObjectKeyFromObject(account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get name and namespace of Acccount object")
+	}
+
+	err = r.Client.Get(context.TODO(), accountObjectKey, account)
+	if err != nil {
+		reqLogger.Error(err, "Unable to get updated Acccount object after status update")
```
Comment (jharrington22) [inline pkg/controller/account/credentials_rotator.go]: We should be more descriptive here

File: pkg/controller/account/credentials_rotator.go (hunk 1 above)
Comment (rafael-azevedo) [reply pkg/controller/account/credentials_rotator.go]: we are not

**Pull request merged**